laplace/subnetlaplace.py

```diff
@@ -117,14 +111,66 @@ def prior_precision_diag(self):
 
         else:
             raise ValueError('Mismatch of prior and model. Diagonal or scalar prior.')
+    
+    @property
+    def mean_subnet(self):
```

Comment:
One issue that was not introduced in this PR, but I just thought about it: the `mean` property should maybe be `mean_subset`, since "mean" implies a distribution and, by definition, in subnet LA we only have a distribution over the subset. Hence, "mean" is only meaningful in the context of the subset. I know this is mostly semantics, but could actually confuse the user, e.g. when calling `subnet_la.mean` and `subnet_la.posterior_covariance` the dimensions won't match.

Reply:
Thanks for looking over this, Runa! Yes, this is a good point. I've also thought about this, and have considered the solution you proposed, but that would have made some parts of the subnet laplace implementation more complex (can't remember what exactly). But I agree that it makes more sense conceptually and am happy to change this -- I'll open a separate issue for this!